Sources/Client/DeltaClientApp.swift

```diff
@@ -3,16 +3,18 @@ import DeltaCore
 
 /// The entry-point for Delta Client.
 struct DeltaClientApp: App {
-  // MARK: Global state
-  // These are static so that they can be used from static functions like `modalError`. And because otherwise they can't be captured by the async startup task.
+  /// Resource pack loaded on startup
+  static var loadedResourcePack: LoadedResources?
```

Comment:
why have you changed it to be like this? i don't like that it's an optional, that's why i put it in the state enum. because the resource pack is only ever accessed when accessing the state anyway.

Reply:
I believe it wasn't there in the beginning and that's a variable I've introduced myself
Right now, after the initial startup process is completed, if one updates LoadingState back again to .loading or .loadingWithMessage, it follows that eventually you'd need to call .done(LoadedResources), but LoadedResources wasn't publicly accessible anywhere
As LoadedResources is something entirely initialised on startup after the loading tasks are done, the choice was between making it optional or force-unwrapping it: I chose the former as I deemed it safer

A quick hack around this would have been to pop the state back to the last valid .done state, but I didn't quite like it

Reply:
If you've got a more elegant solution to this lmk

Reply:
before it was an associated value in .done as you said. and it was ok that going to back to .loading would remove any reference of the resources because you shouldn't ever have to change the state to something other than done once the app has loaded

Reply:
I think I misunderstood the purpose of the loadinState then, I thought it was something more loosely coupled that could be triggered whenever to show a loading screen, even after startup
IMO it would be better to eventually refactor this so that a loading event could be safely initiated at any given moment without relying on resources only available at startup, but for now I will remove the optional (and the property itself) as requested and introduce a new loading state, while still keeping the old one, that can be triggered throughout the whole app's lifecycle

Reply:
Renaming loadingState to startupState and introducing a new state "loadingState" which can be invoked from anywhere inside the app
I think this is overall the best solution